### PR TITLE
refactored the github hunk highlighting 

### PR DESCRIPTION
**Pull request by avikalpg**
- Better variable names
- More readable code structure.

## Summary by CodeRabbit

- **Refactor**
  - Improved the performance and readability of the code responsible for highlighting lines in GitHub diff views.
  - Enhanced error handling and logging for the highlighting functionality to aid in debugging.

- **Style**
  - Standardized code style by adding missing semicolons for consistency.

- **Bug Fixes**
  - Fixed an issue with missing file addresses during error handling in the highlighting process.
  - Optimized the logic for identifying and highlighting relevant lines in different diff views.

- **Documentation**
  - Updated variable names for better clarity and consistency across the codebase.

### CODE REVIEW

**Comment by tapishr** [comment]
Is this tested?

**Review by tapishr — APPROVED** [review]

**Comment by avikalpg** [comment]
> Is this tested?

This has been tested now.

File: scripts/highlighting.js [hunk 1]
```diff
@@ -183,80 +183,95 @@ async function FilesInPrBitbucket(response) {
...
+	let isUnifiedDiffView = false;
+	const allInputTagsWithValueAttr = document.querySelectorAll('input[value]'); // for reducing DOM queries in foreach loop.
+	for (const item of allInputTagsWithValueAttr) {
+		const valueAttrOfInputTag = item.getAttribute('value');
+		const checkedAttrOfInputTag = item.getAttribute('checked');
+
+		if (valueAttrOfInputTag === 'unified' || valueAttrOfInputTag === 'split') {
+			if (checkedAttrOfInputTag === 'checked' && valueAttrOfInputTag === 'unified') {
+				isUnifiedDiffView = true; // for unified view
+			}
```
[the remark this answers is not in the document]
Comment (avikalpg) [reply scripts/highlighting.js:196]: This is incorrect. There is attribute named `checked` that we want.

File: scripts/highlighting.js [hunk 2]
```diff
@@ -183,80 +183,95 @@ async function FilesInPrBitbucket(response) {
...
+		}
+	};
```
Comment (MuskanPaliwal) [inline scripts/highlighting.js:198]: could have done in a never nesting way, LOW LOGAF

**Review by MuskanPaliwal — APPROVED** [review]
Please address the single comment I had left, otherwise, it looks good to me

File: scripts/highlighting.js (hunk 2 above)
Comment (avikalpg) [reply scripts/highlighting.js:198]: Yes, I could have made this a single conditional by removing the first condition, but I have kept it to show the thought process in the code: we first find the input field that has the relevant information and then choose the value.

I don't think never nesting is actually possible here. There is no error condition here. It is an if-else system.

**Pull request merged**